### ISSUE TITLE
Can't receive response data when "receiveDataWhenStatusError=true"

### ISSUE THREAD

**Issue by shun5nktm** [flutterchina/dio#624]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |    flutter  |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   1.9.1   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.7   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |   AndroidStudio 3.5.3   |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

I can't receive response data when "receiveDataWhenStatusError=true"
I use protocol buffer on HTTP
But I don't use GRPc
